spring-pulsar-spring-boot-autoconfigure/build.gradle

```diff
@@ -9,11 +9,8 @@ dependencies {
 	annotationProcessor 'org.springframework.boot:spring-boot-autoconfigure-processor'
 	annotationProcessor 'org.springframework.boot:spring-boot-configuration-processor'
 
-	api project (':spring-pulsar')
-	api 'org.springframework.boot:spring-boot'
-	api 'org.springframework.boot:spring-boot-autoconfigure'
-	api 'org.springframework.boot:spring-boot-starter'
-	api 'org.springframework.boot:spring-boot-starter-validation'
+	optional project (':spring-pulsar')
```

Comment:
optional allows direct consumers to pull in only the autoconfigure module and then only when they add spring-pulsar will auto-config kick in.